lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -77,15 +87,30 @@ def schedule=(schedule_hash)
         reload_schedule!
       end
 
+      def setup_schedule(schedule_hash)
```

Comment:
While you're right that bang methods imply that one is more dangerous than the other, I think you can be even more explicit by calling them something like `atomic_setup_schedule` and `racy_fallback_setup_schedule`.